shell/components/SideNav.vue

```diff
@@ -0,0 +1,553 @@
+            addObjects(out, more);
+          } else {
+            const root = more.find((x) => x.name === 'root');
+            const other = more.filter((x) => x.name !== 'root');
+
+            const group = {
+              name:     productId,
+              label:    this.$store.getters['i18n/withFallback'](`product.${ productId }`, null, ucFirst(productId)),
+              children: [...(root?.children || []), ...other],
+              weight:   productMap[productId]?.weight || 0,
+            };
+
+            addObject(out, group);
+          }
+        }
+      }
+    },
+
+    getExplorerGroups(out) {
+      if ( this.isExplorer ) {
+        const allNavLinks = this.allNavLinks;
+        const toAdd = [];
+        const haveGroup = {};
+
+        for ( const obj of allNavLinks ) {
+          if ( !obj.link ) {
+            continue;
+          }
+
+          const groupLabel = obj.spec.group;
+          const groupSlug = obj.normalizedGroup;
+
+          const entry = {
+            name:        `link-${ obj._key }`,
+            link:        obj.link,
+            target:      obj.actualTarget,
+            label:       obj.labelDisplay,
+            sideLabel:   obj.spec.sideLabel,
+            iconSrc:     obj.spec.iconSrc,
+            description: obj.spec.description,
+          };
+
+          // If there's a spec.group (groupLabel), all entries with that name go under one nav group
+          if ( groupSlug ) {
+            if ( haveGroup[groupSlug] ) {
+              continue;
+            }
+
+            haveGroup[groupSlug] = true;
+
+            toAdd.push({
+              name:     `navlink-group-${ groupSlug }`,
+              label:    groupLabel,
+              isRoot:   true,
+              // This is the item that actually shows up in the nav, since this outer group will be invisible
+              children: [
+                {
+                  name:  `navlink-child-${ groupSlug }`,
+                  label: groupLabel,
+                  route: {
+                    name:   'c-cluster-navlinks-group',
+                    params: {
+                      cluster: this.clusterId,
+                      group:   groupSlug,
+                    }
+                  },
+                }
+              ],
+              weight: -100,
+            });
+          } else {
+            toAdd.push({
+              name:     `navlink-${ entry.name }`,
+              label:    entry.label,
+              isRoot:   true,
+              // This is the item that actually shows up in the nav, since this outer group will be invisible
+              children: [entry],
+              weight:   -100,
+            });
+          }
+        }
+
+        addObjects(out, toAdd);
+      }
+    },
+
+    groupSelected(selected) {
+      this.$refs.groups.forEach((grp) => {
+        if (grp.canCollapse) {
+          grp.isExpanded = (grp.group.name === selected.name);
+        }
+      });
+
+      // this.$emit('expand', selected);
```

Comment:
to remove, not in default layout groupSelected

Reply:
Don't know where this one came from... 🤔